### DISCUSSION TITLE
I need help

### DISCUSSION THREAD

**Discussion by What7268** [Prince25/StockAlertBot#52] [Q&A] [answered]
I dont understand the prerequisites. I got Node, I downloaded the link, it says go to the root directory of stockalertbot and i assume its what i just downloaded. I dont understand npm packages via terminal npm install. I don't know what npm is and I dont see an install button anywhere sorry. I did the next steps in text editor but no clue how to start the program.

**Comment by What7268** [comment]
Oh shoot i realized i needed to go to cmd and type npm install. it gave me warnings and i think nothing happened. says no such file open C:\windows \system32\package.json

**Comment by What7268** [comment]
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path C:\WINDOWS\system32\package.json
npm ERR! errno -4058
npm ERR! enoent ENOENT: no such file or directory, open 'C:\WINDOWS\system32\package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Marc\AppData\Roaming\npm-cache\_logs\2021-03-13T05_58_36_738Z-debug.log

**Reply by What7268** [reply]
this happens when i type npm start

**Reply by Jeisonx** [reply]
Hey it looks like you're running the `npm start` command in the wrong place, you will first need to open a Terminal window (CMD) and from CMD you will need to `cd` into the directory where you download and or cloned the project, after that the next step will be to run the command `npm install` and than you will need to run the command `npm start` in the same CMD window.

**Reply by What7268** [reply]
Ay thank you very much. but now i get this when i type npm start 

C:\Users\Marc\Desktop\StockAlertBot-main>npm start

> stock-alert-bot@3.0.0 start C:\Users\Marc\Desktop\StockAlertBot-main
> node main.js

internal/modules/cjs/loader.js:1106
    module.exports = JSONParse(stripBOM(content));
                     ^

SyntaxError: C:\Users\Marc\Desktop\StockAlertBot-main\config.json: Unexpected token h in JSON at position 15
    at parse (<anonymous>)
    at Object.Module._extensions..json (internal/modules/cjs/loader.js:1106:22)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at file:///C:/Users/Marc/Desktop/StockAlertBot-main/main.js:34:5
    at ModuleJob.run (internal/modules/esm/module_job.js:152:23)
    at async Loader.import (internal/modules/esm/loader.js:166:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! stock-alert-bot@3.0.0 start: `node main.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the stock-alert-bot@3.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Marc\AppData\Roaming\npm-cache\_logs\2021-03-13T06_17_52_207Z-debug.log

C:\Users\Marc\Desktop\StockAlertBot-main>

**Reply by Jeisonx** [reply]
It looks like it's having issues reading the config.json file I believe on line 15. Did you manually edit the config.json file? would you mind sharing your config.json file?

**Reply by What7268** [reply]
{
    "URLS": [https://www.bestbuy.ca/en-ca/product/asus-tuf-gaming-nvidia-geforce-rtx-3080-10gb-gddr6x-video-card/14953248],
    "SUPPORTED_STORES_DOMAINS": [
        "amazon",
        "antonline",
        "argos",
        "bestbuy",
        "costco",
        "currys",
        "gamestop",
        "microcenter",
        "newegg",
        "target",
        "tesco",
        "tescopreorders"
    ],
    "INTERVAL": {
        "unit": "seconds",
        "value": 2700
    },
    "OPEN_URL": true,
    "ALARM": true,
    "AMAZON_DELAY": 10,
    "TARGET_ZIP_CODE": "90024",
    "TARGET_KEY": "ff457966e64d5e877fdbad070f276d18ecec4a01",
    "WEBHOOK_URLS": [],
    "SUPPORTED_WEBHOOK_DOMAINS": [
        "discord",
        "slack"
    ],
    "PROXIES": false,
    "EMAIL": true,
    "SMS_METHOD": "None"
}

**Reply by Jeisonx** [reply]
Your best buy URL should be inside of "" marks after the parenthesis and before closing the parenthesis for example ["https://www.bestbuy.ca/en-ca/product/asus-tuf-gaming-nvidia-geforce-rtx-3080-10gb-gddr6x-video-card/14953248"]

**Reply by What7268** [reply]
ah okay cool thanks! i believe it is working now, i did npm start and it says

C:\Users\Marc\Desktop\StockAlertBot-main>npm start

> stock-alert-bot@3.0.0 start C:\Users\Marc\Desktop\StockAlertBot-main
> node main.js

Does my computer need to be like on/active 24/7 for this to keep working or can i shut my pc off and somehow this tracker still tracks?

**Reply by Jeisonx** [reply]
It looks like it running if the item you're looking for is not in stock it will continue to monitor that url at the interval you see. If you turn off your pc it will stop the project from working and it will not track the url.

**Reply by What7268** [reply]
ah okay thanks for your help! and thanks for making this

**Reply by Jeisonx** [reply]
You're welcome good to hear it's working. I didn't make this all credit goes to @PrinceS25 I'm just here to help let me know if you have any more questions and or issues. And please share and consider donating if this project helps you get the item you want.

**Comment by Prince25** [comment, marked as answer]
Solved here: https://github.com/PrinceS25/StockAlertBot/discussions/52#discussioncomment-477193

**Comment by What7268** [comment]
Sorry I get this error i believe frequently. 

TypeError: Cannot read property 'childNodes' of undefined
    at Timeout.bestbuy [as _onTimeout] (file:///C:/Users/Marc/Desktop/StockAlertBot-main/stores/bestbuy.js:53:67)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

I have no idea what that means

**Reply by Prince25** [reply]
Does that stop the program? If so, try out the latest update (c2c2ed7). You can copy over your `config.json` file. If issue persists, please provide me with the urls you're checking.
